doc/man3/SSL_CTX_set1_curves.pod

```diff
@@ -75,17 +75,76 @@ to downgrade connections to a weaker algorithm.
+group listed will also be used for the B<key_share> sent by a client in a
+TLSv1.3 B<ClientHello>. For servers note the discussion above. The list should
+be in order of preference with the most preferred group first.
+
+Group tuples of comparable security are defined by wrapping such groups with B<{..}>
+brackets. Key shares to be sent by a client are defined by preceding the group
+name with C<*>, for example
+{P-521:*P-256}:{*P-384}:{*X25519:P-384:ffdhe2048}.
+General syntax for the string B<list> is:
+C<[{] [-][?][*]group1 rep[:[-][?][*]groupN] [}] rep[: {[-][?][*]groupM rep[:[-][?][*]groupP]} ]>
+
+If a group name is preceded with the C<?> character, it will be ignored if an
+implementation is missing. If a group name is preceded with the C<-> character, it
+will removed from the list of groups if present, ignored otherwise. The "DEFAULT"
+group name can be used to select the OpenSSL built-in default list of groups.
+
+For a TLS 1.3 client, the groups are used directly in the supported groups extension, any
+`{}` is ignored. The extension's preference order, to be evaluated by the server,
+is determined by the order of the elements in the array. If a group name is preceded
+by C<*>, a key share will be sent for this group. A maximum of 4 key shares are supported.
+If no C<*> is used anywhere in the list, a single key share for the leftmost group is sent.
+
+For a TLS 1.3 server, the groups in string B<list> will be used to determine which group
+is used for the key agreement. The preference order of the group tuples, to be evaluated
+by the server, is determined by the order of the tuples in the array and the preference
+order of the groups within a group tuple is determined by the order of the groups
+in the tuple. Any C<*> will be ignored by a server. Server preference can be enforced
+by using C<SSL_OP_CIPHER_SERVER_PREFERENCE> (default: client preference).
+
+The server will select the group to be used for a key agreement using the following
+pseudo-code algorithm:
+
+ FOR each group tuple
+     IF client preference (= default)
+         FOR each client key-share group
+             IF current key-share group is also part of current group tuple: SH, return success
+         FOR each client supported groups
+             IF current supported group is also part of current group tuple: HRR, return success
+     ELSE (= server preference = with SSL_OP_CIPHER_SERVER_PREFERENCE option set)
+         FOR each group in current tuple
+             IF current group is also part of client key-share groups: SH, return success
+             IF current group is also part of client supported groups: HRR, return success
+ return failure
+
+ with : SH:  Server hello with current group
+        HRR: Server retry request with current group
+
+Hence, if a client supports a group in a server group tuple, a HRR is triggered if
+the client has not also sent a key share for this group. For the above example, if a
```

Comment:
The reader has long lost track of the "above example", it should likely be repeated.

Reply:
Great point, I've repeated the example